src/rmkit/input/events.cpy

```diff
@@ -89,20 +92,27 @@ namespace input:
     public:
     int x=-1, y=-1
     int slot = 0, left = -1
+    int fingers = 0
+    bool lifted=false
+    static int MAX_SLOTS
     struct Point:
       int x=-1, y=-1, left=-1
     ;
+
     vector<Point> slots;
     TouchEvent():
-      slots.resize(10)
+      slots.resize(MAX_SLOTS)
 
     TouchEvent(const TouchEvent &t):
       self.slot = t.slot
       self.slots = t.slots
       self.x = t.x
       self.y = t.y
       self.left = t.left
+      self.lifted = t.lifted
```

Comment:
I'm still seeing fingers=0 in a SynMotionEvent handler when going through `dynamic_cast<TouchEvent*>(ev.original().get())->fingers`. I think it's b/c that isn't copied here. Does this need an explicit copy constructor, or would the default work?

Reply:
i removed explicit constructor, hopefully things improve, but it might be that count_fingers() is always useful to call before checking fingers (and we put that in to main_loop?).